src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,215 @@
+    });
+  }
+
+  /**
+   * Sets the charge calculation type and charge time type data
+   */
+  setChargeForm() {
+    this.chargeForm.get('chargeAppliesTo').valueChanges.subscribe((chargeAppliesTo) => {
+      switch (chargeAppliesTo) {
+        case 1:
+          this.chargeCalculationTypeData = this.chargesTemplateData.loanChargeCalculationTypeOptions;
+          this.chargeTimeTypeData = this.chargesTemplateData.loanChargeTimeTypeOptions;
+          break;
+        case 2:
+          this.chargeCalculationTypeData = this.chargesTemplateData.savingsChargeCalculationTypeOptions;
+          this.chargeTimeTypeData = this.chargesTemplateData.savingsChargeTimeTypeOptions;
+          break;
+        case 3:
+          this.chargeCalculationTypeData = this.chargesTemplateData.clientChargeCalculationTypeOptions;
+          this.chargeTimeTypeData = this.chargesTemplateData.clientChargeTimeTypeOptions;
+          break;
+        case 4:
+          this.chargeCalculationTypeData = this.chargesTemplateData.shareChargeCalculationTypeOptions;
+          this.chargeTimeTypeData = this.chargesTemplateData.shareChargeTimeTypeOptions;
+          break;
+      }
+    });
+  }
+
+  /**
+   * @returns {any} Filtered charge calculation type data.
+   */
+  filteredChargeCalculationType(): any {
+    return this.chargeCalculationTypeData.filter((chargeCalculationType: any) => {
+      if (this.chargeForm.get('chargeTimeType').value === 12 && (chargeCalculationType.id === 3 || chargeCalculationType.id === 4)) {
+        return false;
+      }
+      if (this.chargeForm.get('chargeTimeType').value !== 12 && chargeCalculationType.id === 5) {
+        return false;
+      }
+      if (this.chargeForm.get('chargeAppliesTo').value === 2) {
+        if (!(this.chargeForm.get('chargeTimeType').value === 5 || this.chargeForm.get('chargeTimeType').value === 16) && chargeCalculationType.id === 2) {
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the user form
+   */
+  setConditionalControls() {
+    this.chargeForm.get('chargeAppliesTo').valueChanges.subscribe((chargeAppliesTo) => {
+      switch (chargeAppliesTo) {
+        case 1: // Loan
+          this.chargeForm.addControl('chargePaymentMode', new FormControl('', Validators.required));
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 2: // Savings
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 3: // Client
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.addControl('incomeAccountId', new FormControl(''));
+          break;
+        case 4: // Shares
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          this.chargeForm.get('penalty').setValue(false);
```

Comment:
Set disabled from here itself.

Reply:
This is my mistakes, this should not be set to disabled at all, but as readonly.

Reply:
Readonly does not produce the error we have discussed.

Reply:
No set it as diabled from here and remove the disabled attribute from the html.

Reply:
Readonly doesn't exist for mat-checkbox, I have used disabled as originally, but I have fetched the form raw value, which returns the disabled values too.